src/subpages/list.tsx

```diff
@@ -1,263 +1,138 @@
   return (
-    <div>
+    <main>
       <Checkbox
         label="Include blocks having any tx"
         checked={excludeEmptyTxs}
-        onChange={(_, checked) => {
-          setExcludeEmptyTxs(!!checked);
-        }}
+        onChange={() => setExcludeEmptyTxs(!excludeEmptyTxs)}
       />
-      <BlockListComponent
-        variables={{ offset, limit, excludeEmptyTxs }}
-        pollInterval={POLL_INTERVAL}>
+      <BlockListComponent variables={{ offset, limit, excludeEmptyTxs }}>
         {({ data, loading, error }) => {
-          if (error) return <p>error!</p>;
-
-          const timestamps: Date[] | null =
-            data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => new Date(block!.timestamp))
-              : null;
-
-          let interval: number | null = timestamps ? 0 : null;
-          if (interval != null && timestamps) {
-            for (let i = 0; i < timestamps.length - 1; i++) {
-              interval += +timestamps[i] - +timestamps[i + 1];
-            }
-            interval /= (timestamps.length - 1) * 1000;
+          if (error) {
+            console.error(error);
+            return <p>{error.message}</p>;
           }
-
-          const difficulties: number[] | null =
+          const blocks =
             data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => block!.difficulty)
+              ? (data.blockQuery.blocks as Block[])
               : null;
-          let difficulty = 0;
-          if (difficulty != null && difficulties) {
-            difficulty =
-              difficulties.reduce((d, sum) => d + sum, 0) / difficulties.length;
-          }
 
-          const txNumbers: number[] | null =
-            data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => block!.transactions.length)
-              : null;
-          let totalTxNumber = 0;
-          if (difficulty != null && difficulties) {
-            totalTxNumber = txNumbers.reduce((a, b) => a + b, 0);
-          }
+          const timeTaken: IColumn = {
+            key: 'columnTimeTaken',
+            name: 'Time Taken',
+            minWidth: 50,
+            maxWidth: 200,
+            ...commonProps,
+            isSortedDescending: true,
+            data: 'string',
+            isPadded: true,
+            onRender: (block: Block, index) => {
+              if (blocks === null || index === undefined)
+                throw Error('blocks is null');
+              // FIX TODO: beforeBlock index is not available
+              const beforeBlock =
+                blocks[Math.min(index + 1, blocks.length - 1)];
+              const beforeTimestamp = Date.parse(beforeBlock.timestamp);
+              const nowTimestamp = Date.parse(block.timestamp);
+              return <>{(nowTimestamp - beforeTimestamp) / 1000}</>;
+            },
+          };
```

Comment:
`timeTaken` seems not used anywhere, does it?

Reply:
transaction list render dons't work. so I add this for comment.